### PR TITLE
feat: add contract earnings to all account types

### PR DESCRIPTION
**Pull request by shrimpynuts**
PE-908: https://linear.app/splits/issue/PE-908/add-account-earnings-column-to-split-recipients-list

Makes `ContractEarnings` field available for all `Account`s

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -539,88 +538,86 @@ export function updateWithdrawalAmount(
...
+      )
+      contractEarningsWithdrawal.contractEarnings = contractEarningsId
+      contractEarningsWithdrawal.token = tokenId
+      contractEarningsWithdrawal.amount = ZERO
+    }
+    contractEarningsWithdrawal.amount += amount
+    contractEarningsWithdrawal.save()
+  } else if (contractId == accountId || !contractId) {
+    // Funds were pushed throughs split main
+    // contractId == accountId => split distribution, !contractId => split main withdrawal
```
Comment (mihoward21) [inline src/helpers.ts:584]: I think if `contractId == accountId`, we want to skip this part. That means it's a split distribution, right? So we're not actually performing a withdrawal.

Really the split distribution shouldn't be calling this code. It's a weird leftover thing we never got rid of for legacy reasons. Was a bad design choice in the beginning.

**Comment by mihoward21** [comment]
Can you try deploying this and we'll see if it works? Can just run `yarn build-and-deploy-mainnet-testing`

File: src/helpers.ts (hunk 1 above)
Comment (shrimpynuts) [reply src/helpers.ts:584]: It logically seems a bit strange, but I think we want this?

This is part of the new changes

Split contracts will now have their own contractEarnings (which tracks balances a split earns from other splits/contracts) and distributing a split should move the balance from `internalBalance` => `withdrawals`

Comment (mihoward21) [reply src/helpers.ts:584]: Yes, got it now. ok looks good.

**Review by mihoward21 — APPROVED** [review]
lgtm

**Pull request merged**